lib/drawers/publish-page.vue

```diff
@@ -564,6 +410,112 @@
             store.dispatch('finishProgress');
             store.dispatch('showSnackbar', 'Error archiving page');
           });
+      },
+      restorePageClick() {
+        this.$store.dispatch('openModal', {
+          title: 'Restore Published Version',
+          type: 'type-restore-published-version-modal',
+          data: {
+            name: 'restorePublishedVersion',
+            username: this.$store.state.user.username,
+            onConfirm: () => {
+              this.restorePage();
+            }
+          }
+        });
+      },
+      restorePage() {
+        this.$store.dispatch('currentlyRestoring', true);
+        api.getObject(`${this.$store.state.page.uri}@published.json`).then((publishedPage) => {
+          this.restoreHeadComponents(publishedPage.head).then(() => {
+            this.saveComponents(publishedPage.main, 'main', true).then((components) => {
+              this.loopThroughComponets(components);
+            });
+          });
+        })
+          .catch(() => {
+            this.$store.dispatch('currentlyRestoring', false);
+          });
+      },
+      finishedRestoring() {
+        this.$store.dispatch('showSnackbar', {
+          message: 'Page Restored to Published Version'
+        });
+        this.$store.dispatch('currentlyRestoring', false);
+      },
+      restoreHeadComponents(publishedHeadComponents) {
+        return new Promise((resolve) => {
+          // delete current headComponents in one go, which takes care of removing any components that have been added since last publish & any reordering that may have been done
+          this.$store.dispatch('savePage', { head: [] }).then(() => {
+            this.saveComponents(publishedHeadComponents, 'head').then(() => {
+              resolve();
+            });
+          });
+        });
+      },
+      saveComponents(components, path, forceRender = false) {
+        // Save Components and Rerender in the page
+        return new Promise((resolve) => {
+          const arrComponents = [];
+
+          components.forEach((component) =>{
```

Comment:
Please mind spacing in here
```suggestion
          components.forEach((component) => {
```

Reply:
👍